Post Editing Phase 1: List View and Backend Refactoring

andy-wojciechowski opened the pull request:
The amount of changes in this branch were getting kind of large so I wanted to get an initial PR out before I get too much deeper into post editing. This PR takes care of a few things.

1. Reorged the scss style sheets. Now in the project the stylesheets should look like
stylesheets
  -styles
  -application.scss
  -main.scss

The styles folder would contain stylesheets specific to actual pages in this application. application.scss is required in the application.html.erb file as styles common to our whole application and the main.scss would import any third party styles and our custom stylesheets in the styles subdirectory.

2. Decoupled the post submission logic. GithubService.submit_post was getting really difficult to test so I parked that logic into a new service called the PostService. So, the GithubService has the sole responsibly of only interacting with the GitHub API and delegates responsibly about posts to other classes. 

3. Un-commented out old code which started on this feature and finished up the post list view. 
![Capture](https://user-images.githubusercontent.com/31997141/62000652-c900b200-b0a1-11e9-800f-6962cba46c13.PNG)
It's pretty plain at the moment, but as always I will need help prettying up the page and maybe we can work it into #21 

Note, the actual editing submission logic has not been implemented yet and having the list view only return posts written by the user has not been implemented yet and will be implemented in future PRs.

andy-wojciechowski commented:
Note: this 2nd phase of post editing will be ready for PR once this gets merged. That code is on feature/finishPostEditingPhase2

balloneij commented on app/assets/stylesheets/styles/post-edit.scss:
```diff
@@ -10,12 +10,12 @@ label {
 }
 
-#container {
+#edit-container {
```
This would make more sense as a class.  I know you probably only have one, but that's not really a good reason.  ID's should be used on things that are unique.  An edit-container?  Not so much. The main navbar?  I guess... but. me personally, I wouldn't.  Also this would look silly in HTML.  Now I have to do `<div id="edit-container"></div>`, which suddenly looks terrible when I want to add additional classes `<div id="edit-container" class="hoverable dark-themed"></div>`.  Now think down the road when you suddenly want two for whatever reason.  You're going to have to go back and change this anyway.  Also you can only have one ID so what's the point.

Recap

- When in doubt, use a class
- Think you should use an ID? Prolly not
- I should never see more than a couple ID's used a page
- ID should be used for unique stuff and used sparingly
- Classes should be used for everything because it makes sense.  You can derive from multiple classes
- "But my Javascript selects by ID!!!" Select the root node for your editor by ID, and then grab everything else by class

andy-wojciechowski replied:
This was given the class editContainer and the id was removed.

balloneij replied:
I suppose the last thing I'm wondering is what the filenames mean, and is it inconsistent to have the files name be named-with-dashes or camelCased now that the CSS has camel case?

andy-wojciechowski replied:
My original intent of those filenames was to indicate the controller and view the styles were used on. I know that filename isn't really standard so I renamed the files like below

styles/post/edit.scss
styles/post/list.scss

if you have any other ideas for stylesheet filenames lets tackle that in issue #21

balloneij commented on app/assets/stylesheets/styles/post-edit.scss:
```diff
@@ -10,12 +10,12 @@ label {
 }
 
-#header-container {
+#edit-header-container {
```
Use a class

andy-wojciechowski replied:
This element was given the class editHeaderContainer and the id was removed

balloneij commented on app/assets/stylesheets/styles/post-edit.scss:
```diff
@@ -35,7 +35,7 @@ label {
 }
 
-#middle-container {
+#edit-middle-container {
```
Use a class

andy-wojciechowski replied:
This was given the class editMiddleContainer and the id was removed

balloneij commented on app/assets/stylesheets/styles/post-list.scss:
```diff
@@ -0,0 +1,26 @@
+#create-post-link {
+  float: right;
+  margin-top: 24px;
+}
```
This file is too ID happy.  Use classes.  Also you are missing a newline at the end of the file

andy-wojciechowski replied:
The element was given the class createPostLink as well in other elements in post-list-scss

balloneij commented on app/assets/stylesheets/main.scss:
```diff
@@ -0,0 +1,4 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
+@import "styles/post-edit.scss";
+@import "styles/post-list.scss";
```
Missing newline at the end of file

andy-wojciechowski replied:
I added the missing newline.

balloneij commented on app/assets/stylesheets/application.scss:
```diff
@@ -1,19 +1,6 @@
 
-@import "bootstrap-sprockets";
-@import "bootstrap";
+ @import "main.scss";
```
Missing newline

andy-wojciechowski replied:
I added the missing newline.

balloneij commented on test/integration/post_controller_test.rb:
```diff
@@ -1,54 +1,55 @@
-  #   # Arramge
-  #   setup_session('access token', true)
+  test 'an authenticated user should be able to navigate to / successfully' do 
+    # Arramge
```
Arrange? This is throughout the file

andy-wojciechowski replied:
Yes, as in the Arrange, Act, Assert format for writing unit tests. I put those comments to mark those sections in my automated tests.

balloneij replied:
It's a typo. You have arramge with an M lol.  Looks like it's a copy-paste error throughout the file

andy-wojciechowski replied:
lol I didn't even notice that. You did say Arrange in your original comment so I was confused.

andy-wojciechowski commented:
@balloneij Alright I think I resolved your PR feedback and this should be ready for review again. I removed all ids on containers and replaced them with classes. I left some ids on form controls to make javascript selection easier, but for styling purposes elements should now be mainly relying on classes. I'll keep this in mind going forward. If you have any other suggestions please let me know and I'll get them resolved.

johnwilliamssite commented on app/assets/stylesheets/styles/post-edit.scss:
```diff
@@ -10,16 +10,16 @@ label {
 
-#header-container {
+.editHeaderContainer {
   margin-top: 10px;
```
This kinda goes for the whole file but we should avoid using absolute units for spacing

andy-wojciechowski commented:
@balloneij Grace has this setup to dismiss reviews when you add additional commits so when I resolved you additional comments it dismissed your approval. I've messaged Grace to approve this PR but keep that in mind for the future.

rgeorgebu commented on app/assets/stylesheets/styles/post-edit.scss:
```diff
@@ -10,16 +10,16 @@ label {
 
-#header-container {
+.editHeaderContainer {
   margin-top: 10px;
```
kinda goes for the whole file but you want to avoid non-relative unit stying

andy-wojciechowski replied:
Why? We're targeting this to be used on full resolution desktops to be used by the eboard and members we trust to submit posts. This is not going to be a mobile friendly app especially with the drag and drop image behavior. Anyway with this PR merged please take the non-relative unit styling discussion to Issue #21 so we can maybe include this once we make styling improvements to the post editor.
